### PR TITLE
Ignore the channel we're talking in

### PR DESCRIPTION
**Pull request by br0xen**
Resolves issue #1

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -26,7 +26,10 @@ func main() {
 
 	api := slack.New(cnf.SlackToken)
 
-	c, err := selectChannel(coll, api)
+	var ignore Channels
+	ignore = append(ignore, Channel{Name: cnf.DestChan})
```
Comment (jcbwlkr) [inline main.go]: Let's get this on one line by using a literal

```
ignore := Channels{Channel{Name: cnf.DestChan}}
```

You could go crazy and leave off the field name like

```
ignore := Channels{Channel{cnf.DestChan}}
```

And when you have a slice of a thing you don't have to specify the thing type every time so you could leave off `Channel` too

```
ignore := Channels{{Name: cnf.DestChan}}
```

Or go super nuts

```
ignore := Channels{{cnf.DestChan}}
```

But that's crazy :)

File: select.go [hunk 2]
```diff
@@ -34,11 +35,22 @@ func selectChannel(coll *mgo.Collection, api *slack.Client) (slack.Channel, erro
...
 	// Every channel has already been announced. Truncate the collection then
-	// send a random one.
+	// send a random one that isn't in the 'ignored' collection.
 	log.Print("Couldn't find a new one")
 	if err := coll.DropCollection(); err != nil {
 		return slack.Channel{}, err
 	}
 
-	return all[rand.Intn(len(all))], nil
+	for len(all) > 0 {
```
Comment (jcbwlkr) [inline select.go:44]: Won't this infinite loop if `ignore` == `all`? We don't shorten `all` when we skip.

Comment (br0xen) [reply select.go:44]: Shoot, you're right, I accidentally dropped the line that removes the channel we tried.

File: main.go (hunk 1 above)
Comment (br0xen) [reply main.go]: Yeah, I was conflicted on the initialization of `ignore`, I wasn't sure if we'd want to add some stuff in the DB or something to populate it and then just append the `DestChan` to it or what. But it probably makes more sense to initialize it with `DestChan`, then if we need to add more to it later we can take care of that then.

Comment (br0xen) [reply main.go]: There's not a real good reason to break it down farther than `ignore := Channels{Channel{Name: cnf.DestChan}}`, is there? Going more makes the code smaller, but I feel like this point is the most easily readable for anyone not too familiar with the project. It clearly states what all is going on.

Comment (jcbwlkr) [reply main.go]: Yeah this form is probably best.

**Pull request merged**